src/modules/QtQml/Component.js

```diff
@@ -38,6 +38,10 @@ QMLComponent.prototype.createObject = function(parent, properties, componentCont
     engine.$basePath = bp;
 
     engine.operationState = oldState;
+
+    if (item.hasOwnProperty("parent") && parent) // Seems to be an Item. TODO: Use real inheritance and ask using instanceof.
```

Comment:
`item instanceof getConstructor('QtQuick.Item')` (where getConstructor needs to be replaced by whatever function you can get the constructor function of a class nowadays) should work by now. Would be cool, if you could verify that and use instanceof if it works. :)